lib/slots.ts

```diff
@@ -313,3 +362,19 @@ export function SetSlot<KeyT extends SlotKey>(
 
   slots[id] = value;
 }
+
+export function ResetSlot<KeyT extends SlotKey>(
+  container: DateTimeFormatImpl,
+  id: KeyT,
+  value: Slots[KeyT]['value']
+): void {
+  const slots = GetSlots(container);
+
+  if (slots === undefined) throw new TypeError('Missing slots for the given container');
+
+  const existingSlot = slots[id];
+
+  if (existingSlot === undefined) throw new TypeError(`tried to reset ${id} which was not set`);
+
+  slots[id] = value;
+}
```

Comment:
This is new because we made SetSlot in this polyfill work only once per object per slot, which makes sense because Temporal objects are immutable. But the Intl.DateTimeFormat monkeypatch uses lazily computed slots, because of the slow creation time of Intl.DateTimeFormat objects, which doesn't fit with that. So I added ResetSlot and intend to only use it in getSlotLazy.